rotkehlchen/chain/evm/decoding/curve_lend/utils.py

```diff
@@ -0,0 +1,133 @@
+
+    try:
+        if last_vault_count is not None and vault_count == int(last_vault_count):
+            log.debug(f'Same number ({vault_count}) of Curve vaults returned from API as previous query. Skipping vault processing.')  # noqa: E501
+            return
+    except ValueError:
+        log.error(f'Failed to check last Curve vault count due to {last_vault_count} not being an int')  # noqa: E501
+        return
+
+    for vault in vault_list:
+        try:
+            if vault['blockchainId'] == 'fraxtal':
+                continue
+            elif vault['blockchainId'] == 'arbitrum':
+                vault_chain_id = ChainID.ARBITRUM_ONE
+            else:
+                vault_chain_id = ChainID.deserialize_from_name(vault['blockchainId'])
+
+            underlying_token = get_or_create_evm_token(
+                userdb=database,
+                evm_address=string_to_evm_address(vault['assets']['borrowed']['address']),
+                chain_id=vault_chain_id,
+                decimals=deserialize_int(vault['assets']['borrowed']['decimals']),
+                symbol=vault['assets']['borrowed']['symbol'],
+                encounter=TokenEncounterInfo(description='Querying Curve vaults', should_notify=False),  # noqa: E501
+            )
+            vault_token = get_or_create_evm_token(
+                userdb=database,
+                evm_address=string_to_evm_address(vault['address']),
+                chain_id=vault_chain_id,
+                protocol=CURVE_VAULTS_PROTOCOL,
+                name=vault['name'],
+                symbol='cvcrvUSD',
+                underlying_tokens=[UnderlyingToken(
+                    address=underlying_token.evm_address,
+                    token_kind=EvmTokenKind.ERC20,
+                    weight=ONE,
+                )],
+                encounter=TokenEncounterInfo(description='Querying Curve vaults', should_notify=False),  # noqa: E501
+            )
+
+            # Cache the controller and AMM addresses to avoid having to make a call
+            # to the vault contract when they're needed during decoding.
+            with GlobalDBHandler().conn.write_ctx() as write_cursor:
+                globaldb_set_unique_cache_value(
+                    write_cursor=write_cursor,
+                    key_parts=[CacheType.CURVE_VAULT_CONTROLLER, vault_token.evm_address],
+                    value=vault['controllerAddress'],
+                )
+                globaldb_set_unique_cache_value(
+                    write_cursor=write_cursor,
+                    key_parts=[CacheType.CURVE_VAULT_AMM, vault_token.evm_address],
+                    value=vault['ammAddress'],
+                )
+
+        except (DeserializationError, KeyError) as e:
+            error = f'missing key {e!s}' if isinstance(e, KeyError) else f'{e!s}'
+            log.error(f'Failed to store token information for Morpho vault due to {error}. Vault: {vault}. Skipping...')  # noqa: E501
```

Comment:
See? You are just copy-pastaing.

Please **do not** do that and always favor abstraction. Writing code is not just copying other code.